tyssue/collisions/solvers.py

```diff
@@ -40,6 +45,57 @@ def with_collision_correction(*args, **kwargs):
     return with_collision_correction
 
 
+def solve_self_intersect_face(eptm):
+    face_self_intersect = eptm.edge_df.groupby("face").apply(_do_face_self_intersect)
+
+    for f in face_self_intersect[face_self_intersect].index:
+        sorted_edge = np.array(_ordered_edges(
+            eptm.edge_df[eptm.edge_df['face'] == f][['srce', 'trgt', 'face']])).flatten()[3::4]
+        angle_list = np.arctan2(
+            eptm.edge_df.loc[sorted_edge]['sy'].to_numpy() - eptm.edge_df.loc[sorted_edge]['fy'].to_numpy(),
+            eptm.edge_df.loc[sorted_edge]['sx'].to_numpy() - eptm.edge_df.loc[sorted_edge][
+                'fx'].to_numpy()) * 180 / np.pi + 180
+
+        angle_e = pd.DataFrame(angle_list, index=sorted_edge, columns=['angle'])
+
+        if np.where(np.min(angle_e['angle']) == angle_e['angle'])[0] != 0:
+            pos_s = np.where(np.min(angle_e['angle']) == angle_e['angle'])[0][0]
+            angle_e = pd.concat([angle_e.iloc[pos_s:], angle_e.iloc[:pos_s]])
+
+        angle_e = pd.concat([angle_e, angle_e.iloc[[0]]])
+        angle_e.iloc[-1]['angle'] += 360
+
+        pos_s = np.where(angle_e.diff()['angle'] < 0)[0][0]
+        v1 = eptm.edge_df.loc[angle_e.index[pos_s]]['srce']
+        v2 = eptm.edge_df.loc[angle_e.index[pos_s - 1]]['srce']
+        if eptm.vert_df.loc[v1]['segment'] == eptm.vert_df.loc[v2]['segment']:
```

Comment:
why is this dependent on the `segment` column? It is not always there?

Reply:
I still had some problem that I try to understand. Code now is working only for 2D lateral geometry cause of the `segment` column. I will remove it.